src/data/java/com/github/minecraftschurlimods/arsmagicalegacy/data/AMRecipeProvider.java

```diff
@@ -414,18 +414,21 @@ protected void buildCraftingRecipes(Consumer<FinishedRecipe> consumer) {
                 .define('P', Items.PAPER)
                 .unlockedBy("has_paper", has(Items.PAPER))
                 .save(consumer);
-        ShapelessRecipeBuilder.shapeless(AMItems.MANA_CAKE.get(), 3)
-                .requires(Tags.Items.CROPS_WHEAT)
-                .requires(Items.SUGAR)
-                .requires(AMItems.AUM.get())
-                .requires(AMItems.CERUBLOSSOM.get())
+        ShapedRecipeBuilder.shaped(AMItems.MANA_CAKE.get())
+                .pattern("AC")
+                .pattern("SE")
+                .pattern("MM")
+                .define('A', AMItems.AUM.get())
+                .define('C', AMItems.CERUBLOSSOM.get())
+                .define('S', Items.SUGAR)
+                .define('E', Tags.Items.EGGS)
+                .define('M', Items.MILK_BUCKET)
                 .unlockedBy("has_cerublossom", has(AMItems.CERUBLOSSOM.get()))
                 .save(consumer);
         ShapelessRecipeBuilder.shapeless(AMItems.MANA_MARTINI.get())
-                .requires(Tags.Items.RODS_WOODEN)
-                .requires(Tags.Items.CROPS_POTATO)
+                .requires(AMTags.Items.DUSTS_VINTEUM)
+                .requires(Items.BAMBOO)
                 .requires(Items.ICE)
-                .requires(Items.SUGAR)
                 .requires(NBTIngredient.of(PotionUtils.setPotion(new ItemStack(Items.POTION), Potions.WATER)))
```

Comment:
why change this recipe?

Reply:
To more closely resemble the new texture (which has a bamboo-like straw), also because not having a magic item in a magic potion-like item seems weird